src/tools/chip-cert/CertUtils.cpp

```diff
@@ -310,6 +330,108 @@ bool AddExtension(X509 * cert, int extNID, const char * extStr)
     return res;
 }
 
+bool SetBasicConstraintsExtension(X509 * cert, bool isCA, int pathLen, CertStructConfig & certConfig)
+{
+    if (certConfig.IsExtensionBasicPresent())
+    {
+        std::string basicConstraintsExt;
+
+        if (certConfig.IsExtensionBasicCriticalPresent())
```

Comment:
use && for these to make condition depth smaller.

Reply:
done